addons/mail/models/discuss/discuss_channel.py

```diff
@@ -102,14 +102,17 @@ def _constraint_group_id_channel(self):
 
     # COMPUTE / INVERSE
 
-    @api.depends('channel_type', 'is_member')
+    @api.depends('channel_type', 'is_member', 'group_public_id')
+    @api.depends_context('uid')
     def _compute_is_editable(self):
+        if not self.check_access_rights('write', raise_exception=False):
+            self.update({'is_editable': False})
```

Comment:
Just assign directly, update is kinda useless/deprecated especially for one field.